src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/ReferenceConfidenceModel.java

```diff
@@ -189,6 +189,10 @@ public ReferenceConfidenceModel(final SampleList samples,
         final int ploidy = ploidyModel.samplePloidy(0); // the first sample = the only sample in reference-confidence mode.
 
         final SimpleInterval refSpan = activeRegion.getSpan();
+        // Ensuring that if the underlying reads have any cached indel informativeness data it gets purged before calling the next region.
+        if (USE_CACHED_READ_INDEL_INFORMATIVENESS_VALUES) {
+            readLikelihoods.sampleReads(0).forEach(r -> r.clearTransientAttribute(INDEL_INFORMATIVE_BASES_CACHE_ATTRIBUTE_NAME));
```

Comment:
Is `sampleReads(0)` correct here? Do we have to worry about the multi-sample case in this context?

Reply:
This behavior is acceptable by virtue of the fact that above there is the following line: `Utils.validateArg(readLikelihoods.numberOfSamples() == 1, () -> "readLikelihoods must contain exactly one sample but it contained " + readLikelihoods.numberOfSamples());` Furthermore when we create pileups on the next in `AssemblyBasedCallerUtils.getPileupsOverReference()` this is the exact mechanism by which we access the reads.